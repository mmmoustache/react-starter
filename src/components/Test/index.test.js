import React from 'react';
import Test from '../Test';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Component: Test', () => {
  let component;

  it('Should render successfully', () => {
    component = shallow(
      <Test updateTerm={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });

  it('updateTerm should be called', () => {
    component.find('button').simulate('click');
    expect(component).toMatchSnapshot();
  });

});
