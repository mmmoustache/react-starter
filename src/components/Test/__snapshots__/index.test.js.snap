// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Test Should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Test
    activeTerm={undefined}
    updateTerm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Hello there!
        </p>,
        <button
          onClick={[Function]}
          type="button"
        >
          Click Me!
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello there!",
        },
        "ref": null,
        "rendered": "Hello there!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click Me!",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Click Me!",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Hello there!
          </p>,
          <button
            onClick={[Function]}
            type="button"
          >
            Click Me!
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello there!",
          },
          "ref": null,
          "rendered": "Hello there!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click Me!",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Click Me!",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Component: Test updateTerm should be called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Test
    activeTerm={undefined}
    updateTerm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Hello there!
        </p>,
        <button
          onClick={[Function]}
          type="button"
        >
          Click Me!
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello there!",
        },
        "ref": null,
        "rendered": "Hello there!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click Me!",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Click Me!",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Hello there!
          </p>,
          <button
            onClick={[Function]}
            type="button"
          >
            Click Me!
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello there!",
          },
          "ref": null,
          "rendered": "Hello there!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click Me!",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Click Me!",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
